ddtrace/contrib/pymemcache/trace.py

```diff
@@ -0,0 +1,142 @@
+
+class WrappedClient(wrapt.ObjectProxy):
+    def __init__(self, *args, **kwargs):
+        c = _Client(*args, **kwargs)
+        super(WrappedClient, self).__init__(c)
+
+    def set(self, *args, **kwargs):
+        return self._cmd("set", *args, **kwargs)
+
+    def set_many(self, *args, **kwargs):
+        return self._cmd("set_many", *args, **kwargs)
+
+    def add(self, *args, **kwargs):
+        return self._cmd("add", *args, **kwargs)
+
+    def replace(self, *args, **kwargs):
+        return self._cmd("replace", *args, **kwargs)
+
+    def append(self, *args, **kwargs):
+        return self._cmd("append", *args, **kwargs)
+
+    def prepend(self, *args, **kwargs):
+        return self._cmd("prepend", *args, **kwargs)
+
+    def cas(self, *args, **kwargs):
+        return self._cmd("cas", *args, **kwargs)
+
+    def get(self, *args, **kwargs):
+        return self._cmd("get", *args, **kwargs)
+
+    def get_many(self, *args, **kwargs):
+        return self._cmd("get_many", *args, **kwargs)
+
+    def gets(self, *args, **kwargs):
+        return self._cmd("gets", *args, **kwargs)
+
+    def gets_many(self, *args, **kwargs):
+        return self._cmd("gets_many", *args, **kwargs)
+
+    def delete(self, *args, **kwargs):
+        return self._cmd("delete", *args, **kwargs)
+
+    def delete_many(self, *args, **kwargs):
+        return self._cmd("delete_many", *args, **kwargs)
+
+    def incr(self, *args, **kwargs):
+        return self._cmd("incr", *args, **kwargs)
+
+    def decr(self, *args, **kwargs):
+        return self._cmd("decr", *args, **kwargs)
+
+    def touch(self, *args, **kwargs):
+        return self._cmd("touch", *args, **kwargs)
+
+    def stats(self, *args, **kwargs):
+        return self._cmd("stats", *args, **kwargs)
+
+    def version(self, *args, **kwargs):
+        return self._cmd("version", *args, **kwargs)
+
+    def flush_all(self, *args, **kwargs):
+        return self._cmd("flush_all", *args, **kwargs)
+
+    def quit(self, *args, **kwargs):
+        return self._cmd("quit", *args, **kwargs)
+
+    def set_multi(self, *args, **kwargs):
+        """set_multi is an alias for set_many"""
+        return self._cmd("set_many", *args, **kwargs)
+
+    def get_multi(self, *args, **kwargs):
+        """set_multi is an alias for set_many"""
+        return self._cmd("get_many", *args, **kwargs)
+
+    def _cmd(self, method_name, *args, **kwargs):
+        """Run and trace the given command.
+
+        Any pymemcache exception is caught and span error information is
+        set. The exception is then reraised for the application to handle
+        appropriately.
+        """
+        method = getattr(self.__wrapped__, method_name)
+        with self._span(method_name) as span:
+            try:
+                return method(*args, **kwargs)
+            except MemcacheClientError:
+                (typ, val, tb) = sys.exc_info()
+                span.set_exc_info(typ, val, tb)
+                six.reraise(typ, val, tb)
```

Comment:
this `reraise` is a bit unsafe because `six` is not one of our dependencies. It works because it's a dependency of `pymemcache` but we can't rely on that. I would say to use our internal helper that does the same as `six`: 
```python
from ddtrace.compat import reraise

raise reraise(typ, val, tb)
```

Also I think this code is quite duplicated. Can't we change it in:
```python
try:
    return method(*args, **kwargs)
    except (MemcacheClientError, MemcacheServerError, ...):
        (typ, val, tb) = sys.exc_info()
        span.set_exc_info(typ, val, tb)
        raise reraise(typ, val, tb)
```

Is this code-path well tested? We must fire this exception and double check the span, so it's flagged as error and with the right stacktrace.